bytes/src/main/scala/io/iohk/ethereum/utils/ByteStringUtils.scala

```diff
@@ -61,9 +61,14 @@ object ByteStringUtils {
   }
 
   def concatByteStrings(elements: Iterator[ByteStringElement]): ByteString = {
-    val builder = new mutable.ArrayBuilder.ofByte
-    elements.foreach(el => builder.addAll(el.asByteArray))
-    ByteString(builder.result())
+    val elementList: List[ByteStringElement] = elements.toList
+    val len = elementList.foldLeft(0)(_ + _.len)
+    val array = Array.ofDim[Byte](len)
+    elementList.foldLeft(0) { (index, el) =>
+      System.arraycopy(el.asByteArray, 0, array, index, el.len)
+      index + el.len
+    }
+    ByteString(array)
```

Comment:
naive question: wouldn't a `ArrayBuilder.sizeHint` and `ArrayBuilder.addAll` achieve the same result?

Reply:
Sure, but if we have known array size - perhaps we won't need a builder.

Reply:
true. 
my point is that it would be less code to maintain for the cost of one class allocation. observe that the underlying code is basically the same as what you wrote and quite similar to what `Array.concat` does.

thanks for the response!